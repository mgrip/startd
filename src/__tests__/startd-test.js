// @flow strict

import Startd from "../startd";
jest.mock("../dirname");

test("startd should correctly generate a webpack config from an input path", () => {
  const startd = new Startd("TEST_APP");
  expect(startd.webpackConfig).toMatchSnapshot();
  const startdWithMiddleware = new Startd("TEST_APP", "TEST_MIDDLEWARE");
  expect(startdWithMiddleware.webpackConfig).toMatchSnapshot();
});
