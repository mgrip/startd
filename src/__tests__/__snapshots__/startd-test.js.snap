// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`startd should correctly generate a webpack config from an input path 1`] = `
Array [
  Object {
    "entry": Array [
      "@babel/polyfill",
      "/TEST_DIRNAME/server.js",
    ],
    "mode": "development",
    "module": Object {
      "rules": Array [
        Object {
          "exclude": /node_modules/,
          "loader": "babel-loader",
          "options": Object {
            "presets": Array [
              "@babel/preset-react",
            ],
          },
          "test": /\\\\\\.\\(js\\|jsx\\)\\$/,
        },
      ],
    },
    "name": "server",
    "node": Object {
      "__dirname": true,
    },
    "output": Object {
      "filename": "server.bundle.js",
      "libraryTarget": "commonjs2",
      "path": "/TEST_DIRNAME",
    },
    "plugins": Array [
      DefinePlugin {
        "definitions": Object {
          "APP_PATH": "\\"TEST_APP\\"",
          "BUNDLE_PATH": "\\"http://localhost:8080/app.bundle.js\\"",
          "PORT": 3000,
        },
      },
    ],
    "target": "node",
  },
  Object {
    "entry": Array [
      "@babel/polyfill",
      "/TEST_DIRNAME/client.js",
    ],
    "mode": "development",
    "module": Object {
      "rules": Array [
        Object {
          "exclude": /node_modules/,
          "loader": "babel-loader",
          "options": Object {
            "presets": Array [
              "@babel/preset-react",
            ],
          },
          "test": /\\\\\\.\\(js\\|jsx\\)\\$/,
        },
      ],
    },
    "name": "client",
    "output": Object {
      "filename": "app.bundle.js",
      "path": "/public",
    },
    "plugins": Array [
      DefinePlugin {
        "definitions": Object {
          "APP_PATH": "\\"TEST_APP\\"",
          "BUNDLE_PATH": "\\"http://localhost:8080/app.bundle.js\\"",
          "PORT": 3000,
        },
      },
    ],
  },
]
`;

exports[`startd should correctly generate a webpack config from an input path 2`] = `
Array [
  Object {
    "entry": Array [
      "@babel/polyfill",
      "/TEST_DIRNAME/server.js",
    ],
    "mode": "development",
    "module": Object {
      "rules": Array [
        Object {
          "exclude": /node_modules/,
          "loader": "babel-loader",
          "options": Object {
            "presets": Array [
              "@babel/preset-react",
            ],
          },
          "test": /\\\\\\.\\(js\\|jsx\\)\\$/,
        },
      ],
    },
    "name": "server",
    "node": Object {
      "__dirname": true,
    },
    "output": Object {
      "filename": "server.bundle.js",
      "libraryTarget": "commonjs2",
      "path": "/TEST_DIRNAME",
    },
    "plugins": Array [
      DefinePlugin {
        "definitions": Object {
          "APP_PATH": "\\"TEST_APP\\"",
          "BUNDLE_PATH": "\\"http://localhost:8080/app.bundle.js\\"",
          "MIDDLEWARE_PATH": "\\"TEST_MIDDLEWARE\\"",
          "PORT": 3000,
        },
      },
    ],
    "target": "node",
  },
  Object {
    "entry": Array [
      "@babel/polyfill",
      "/TEST_DIRNAME/client.js",
    ],
    "mode": "development",
    "module": Object {
      "rules": Array [
        Object {
          "exclude": /node_modules/,
          "loader": "babel-loader",
          "options": Object {
            "presets": Array [
              "@babel/preset-react",
            ],
          },
          "test": /\\\\\\.\\(js\\|jsx\\)\\$/,
        },
      ],
    },
    "name": "client",
    "output": Object {
      "filename": "app.bundle.js",
      "path": "/public",
    },
    "plugins": Array [
      DefinePlugin {
        "definitions": Object {
          "APP_PATH": "\\"TEST_APP\\"",
          "BUNDLE_PATH": "\\"http://localhost:8080/app.bundle.js\\"",
          "MIDDLEWARE_PATH": "\\"TEST_MIDDLEWARE\\"",
          "PORT": 3000,
        },
      },
    ],
  },
]
`;
